Src/Controllers/Frontend/WebhookController.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+namespace Plugin\MonduPayment\Src\Controllers\Frontend;
+
+use JTL\DB\DbInterface;
+use JTL\Shop;
+use Plugin\MonduPayment\Src\Helpers\Response;
+use Plugin\MonduPayment\Src\Models\MonduInvoice;
+use Plugin\MonduPayment\Src\Models\MonduOrder;
+use Plugin\MonduPayment\Src\Models\Order;
+use Plugin\MonduPayment\Src\Support\Http\Request;
+use Plugin\MonduPayment\Src\Support\HttpClients\MonduClient;
+
+class WebhookController
+{
+    private MonduClient $monduClient;
+    private Request $request;
+    private MonduInvoice $monduInvoice;
+    private MonduOrder $monduOrder;
+    private Order $order;
+
+    public function __construct(private DbInterface $db)
+    {
+        $this->monduClient = new MonduClient();
+        $this->request = new Request;
+        $this->monduInvoice = new MonduInvoice();
+        $this->monduOrder = new MonduOrder();
+        $this->order = new Order();
+    }
+
+    /**
+     * @return array
+     */
+    public function handleWebhook()
+    {
+        $requestData = $this->request->all();
+
+        $params = [
+            'order_id' => $requestData['order_id'],
+            'order_number' => $requestData['order_uuid'],
+            'invoice_uuid' => $requestData['invoice_uuid'],
+            'order_state' => $requestData['order_state']
+        ];
+
+        switch ($requestData['topic']) {
+            case 'order/confirmed':
+            case 'order/canceled':
+            case 'order/declined':
+            case 'order/pending':
+                return $this->handleOrderStateChanged($params);
+            case 'invoice/created':
+                return $this->handleInvoiceStateChanged($params, 'created');
+            case 'invoice/canceled':
+                return $this->handleInvoiceStateChanged($params, 'canceled');
+            default:
+                return [['error' => 'Unregistered topic'], Response::HTTP_UNPROCESSABLE_ENTITY];
```

Comment:
Lets return 200 here, because BNPL will resend the webhook if it's not 200 status.

Reply:
Done